lib/bios/menus.robot

```diff
@@ -284,12 +284,15 @@ Get Option State
     [Arguments]    ${menu}    ${option}
     ${index}=    Get Index Of Matching Option In Menu    ${menu}    ${option}
     ${value}=    Get Value From Brackets    ${menu}[${index}]
-    IF    '${value}[0]' == 'X'
-        ${state}=    Set Variable    ${TRUE}
-    ELSE IF    '${value}[0]' == ' '
-        ${state}=    Set Variable    ${FALSE}
-    ELSE
-        ${state}=    Set Variable    ${value}
+    ${len}=    Get Length    ${value}
+
+    ${state}=    Set Variable    ${value}
+    IF    ${len} == 1
+        IF    '${value}[0]' == 'X'
+            ${state}=    Set Variable    ${TRUE}
+        ELSE IF    '${value}[0]' == ' '
+            ${state}=    Set Variable    ${FALSE}
+        END
     END
```

Comment:
Does this actually introduce any changes in how the code works?
It seems more complicated but does the same thing. Or is there some issue with comparing strings which didn't work in the original solution?